handler/cantabular_metadata_export.go

```diff
@@ -210,9 +218,9 @@ func (h *CantabularMetadataExport) Handle(ctx context.Context, workerID int, msg
 	return nil
 }
 
-func (h *CantabularMetadataExport) exportTXTFile(ctx context.Context, e *event.CSVCreated, m dataset.Metadata, isPublished bool) (*downloadInfo, error) {
+func (h *CantabularMetadataExport) exportTXTFile(ctx context.Context, e *event.CSVCreated, m dataset.Metadata, isPublished bool, isCustom bool) (*downloadInfo, error) {
```

Comment:
The issue of de-referencing the pointer can he handled by changing the signature of this function to have a `isCustom *bool` This will allow line 179 to simply pass  `filterOutput.Custom` without de-referencing it and not adding the hack code. Similarly pass the pointer to generateTextFilename and update the check in that function to this
```
if isCustom != nil {
	datasetID = "custom"
}
```

Reply:
Sorry yes, check the pointer and if not null, check the value

Reply:
I would prefer it the way I have done it, as the functions above should not really need to know whether the variable is a pointer to a bool or the bool itself, feels right that they should just have the bool itself, with the check ocurring a little earlier than the two points where it is used